src/lib/isStreamPNGAnimation.ts

```diff
@@ -0,0 +1,16 @@
+import { spawn } from 'duplex-child-process';
+import { Stream } from 'stream';
+
+export default async (stream: Stream): Promise<any> => new Promise((resolve, reject) => {
+  const proc = spawn('strings');
+  stream.pipe(proc);
+  const buffer = [];
+  proc.on('data', (chunk) => {
+    buffer.push(chunk);
+  });
+  proc.on('end', () => {
+    const data = Buffer.concat(buffer).toString();
+    const animated = /acTL/.test(data);
```

Comment:
This test could be executed every time a chunk is received from the process. Then the entire result (as I guess it would amount to almost the size of the image) would not have to be buffered to memory.